### ISSUE TITLE
Better search

### ISSUE THREAD

**Issue by n76** [Chocobozzz/PeerTube#2638]
**What happened?**
Enter a search word or words.

**What do you expect to happen instead?**
Only a limited number of videos are found.

**Steps to reproduce:**

1. Search on "railway"

2. Only a few videos found.

3. Using filters, etc. Search on empty string but have filter to search on tag "railway". A larger number of videos is found.

Suggest that in addition to whatever the current search algorithm is, a generic search should also search for tags matching the search words.

**Additional information**
* PeerTube version or URL: 2.1.1
* Browser name/version: n/a
* NodeJS version: n/a

* Link to browser console log if useful:
* Link to server log if useful (journalctl or /var/www/peertube/storage/logs/):

**Comment by DontUseGithub** [comment]
https://github.com/Chocobozzz/PeerTube/issues/1242

**Comment by Chocobozzz** [comment]
Please could you try to reproduce on https://peertube2.cpy.re/?

**Comment by n76** [comment]
> Please could you try to reproduce on https://peertube2.cpy.re/?

I like the ability to enter the UUID in the search. I had thought about suggesting that.

With respect to my original issue above, I don't see any improvement.

**Comment by Chocobozzz** [comment]
Weird, because when I search "railway" I get 7 results.

If I search on "railway" tag, I get get 4 results :thinking:

**Comment by n76** [comment]
Oops. Yes, you are correct. The results are better on that search.

I've just tried a couple more searches on that instance and on my own and your test instance does a better job at returning more relevant videos. This is definitely an improvement.

Question: Will this change the [API for searching](https://docs.joinpeertube.org/api-rest-reference.html#tag/Search)?

**Comment by Chocobozzz** [comment]
> Question: Will this change the API for searching?

No, the search API is stable. We just reworked a little bit the SQL queries behind the search endpoint.

**Comment by n76** [comment]
> > Question: Will this change the API for searching?
> 
> No, the search API is stable. We just reworked a little bit the SQL queries behind the search endpoint.

Good to hear. I can leave the search logic on my Roku channel unchanged. :)
